base/shell/explorer/trayclock.cpp

```diff
@@ -173,34 +173,19 @@ LRESULT CTrayClockWnd::OnThemeChanged()
 
     if (clockTheme)
     {
-        GetThemeFont(clockTheme,
-            NULL,
-            CLP_TIME,
-            0,
-            TMT_FONT,
-            &clockFont);
+        GetThemeFont(clockTheme, NULL, CLP_TIME, 0, TMT_FONT, &clockFont);
 
         hFont = CreateFontIndirectW(&clockFont);
 
-        GetThemeColor(clockTheme,
-            CLP_TIME,
-            0,
-            TMT_TEXTCOLOR,
-            &textColor);
+        GetThemeColor(clockTheme, CLP_TIME, 0, TMT_TEXTCOLOR, &textColor);
 
         if (this->hFont != NULL)
             DeleteObject(this->hFont);
 
         SetFont(hFont, FALSE);
-    }
-    else
-    {
-        /* We don't need to set a font here, our parent will use
-            * WM_SETFONT to set the right one when themes are not enabled. */
-        textColor = RGB(0, 0, 0);
```

Comment:
Why wouldn't it be possible to have the new `GetSysColor(COLOR_BTNTEXT);` made here instead? Naively it would seem that this would allow to get the color initialization done at just one place.

Reply:
Message `WM_SYSCOLORCHANGE` won't reach the child windows.
Message `WM_SYSCOLORCHANGE` just invalidate the window area.
So, getting the system color at `OnPaint` is reasonable.